crates/bevy_window/src/window.rs

```diff
@@ -123,35 +123,30 @@ impl Window {
     }
 
     #[inline]
-    pub fn width(&self) -> u32 {
-        self.width
+    pub fn logical_width(&self) -> f32 {
+        (self.physical_width as f64 / self.scale_factor) as f32
     }
 
-    pub fn scaled_width(&self) -> u32 {
-        (self.width as f64 * self.scale_factor) as u32
-    }
-
-    pub fn scaled_height(&self) -> u32 {
-        (self.height as f64 * self.scale_factor) as u32
+    #[inline]
+    pub fn logical_height(&self) -> f32 {
+        (self.physical_height as f64 / self.scale_factor) as f32
     }
 
     #[inline]
-    pub fn height(&self) -> u32 {
-        self.height
+    pub fn physical_width(&self) -> u32 {
+        self.physical_width
     }
 
-    pub fn set_resolution(&mut self, width: u32, height: u32) {
```

Comment:
Was removing this method intentional? Users should be able to resize the window in code.

Reply:
No, that was not intentional. It is ambiguous to me. I lean toward using physical unit as specifying logical size doesn't necessary map to a legal physical resoultion.

Reply:
The reason we use logical pixels for specifying desired window size (and reporting windowing events) is that it allows people to code assuming a specific resolution. That way the app behaves the same way on "normal" displays and "hi-dpi" displays / devs normally don't need to think about it.

When a user thinks "i want to set the width to X and the height to Y", that should happen in logical pixels.

Reply:
I just pushed an update that restores this call using logical sizes (though still f32)

Reply:
This is a bit of a tangent so I apologize if this is not the right place for this comment.

With such a wide variety of monitor sizes for "low" DPI monitors I don't see the benefit of designing around a particular resolution. For my professional work I run at native resolution in all cases and scale my UI based on the system font size to get something that works well for a particular user's environment. By sticking to native resolution always, its much easier to get clean lines since I don't have to undo the logical transform to understand where my lines are going to fall. I think an argument could be made for removing logical sizes all together, but that is not a topic for this PR.

Reply:
this now takes `u32` like before

Reply:
Yeah thats a good point. I also think that building adaptable ui is the right call, but i do think its a good idea to at least give the option to automatically scale up according to "OS scale factor". Its a "semi-standard" way for the os to report pixel density. Its definitely worth discussing hi-dpi app programming, but yeah we won't block this pr on that :smile: 

 > this now takes u32 like before

Thanks for making the change. I'm honestly sort of opening up to using floats, but id prefer it if we did it in a separate pr. Ideally it would also include a demonstration that including the extra precision produces noticeable differences in behavior. I would expect literally no difference in behavior for integer scale factor values and _maybe_ differences in things like "sub-logical pixel interpolation" for floating point scale factors.